src/test/java/hudson/tasks/junit/CaseResultTest.java

```diff
@@ -222,42 +225,38 @@ public void testAge() throws Exception {
         //Create a job:
         FreeStyleProject p = rule.createFreeStyleProject(projectName);
         p.getPublishersList().add(new JUnitResultArchiver("*.xml"));
-
-        //First build execution:
+        // add builder copying test result file
         p.getBuildersList().add(new TestBuilder() {
             public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
-                build.getWorkspace().child("junit.xml").copyFrom(
-                    getClass().getResource(testResultResourceFile));
+                FilePath junitFile = build.getWorkspace().child("junit.xml");
+                junitFile.copyFrom(getClass().getResource(testResultResourceFile));
+                // sadly this can be flaky for 1ms.... (but hey changing to nano even in core might complicated :))
+                junitFile.touch(System.currentTimeMillis()+1L);
                 return true;
             }
         });
-        FreeStyleBuild b1 = rule.assertBuildStatus(Result.UNSTABLE, p.scheduleBuild2(0).get());
-
-        //First build result analysis:
-        TestResult tr = b1.getAction(TestResultAction.class).getResult();
-        assertEquals(1,tr.getFailedTests().size());
-        CaseResult cr = tr.getFailedTests().get(0);
-        assertEquals(1,cr.getAge()); //First execution, failing test age is expected to be 1
-
 
+        //First build execution
+        {
+            FreeStyleBuild b1 = rule.assertBuildStatus(Result.UNSTABLE, p.scheduleBuild2(0).get());
```

Comment:
Is something being edited here? If so, it is hard to follow what. If you are adding test coverage for the new option, can it be kept to just creating a new test case method?

Reply:
read current version of the tests. there is some bad copy/paste which doesn't make sense.